tensorflow_addons/optimizers/discriminative_layer_training.py

```diff
@@ -0,0 +1,146 @@
+# Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Discriminative Layer Training Optimizer for TensorFlow."""
+
+import tensorflow as tf
+from typeguard import typechecked
+from typing import Union
+from tensorflow.keras.optimizers import Optimizer
+
+# python -m flake8 tensorflow_addons/optimizers/discriminative_layer_training.py
+# python -m black tensorflow_addons/optimizers/discriminative_layer_training.py
+
+
+class FakeVar:
+    def __init__(self, name):
+        # probably can be refactored out
+        self.name = name
```

Comment:
I'm open to alternative ideas. Its just an object to match how we access the name of weight object

Reply:
Shouldn't `tf.Variable` itself have `name` attribute?

Reply:
I wanted to avoid serializing the whole variable as part of the optimizer, so I used the name. This has been refactored to remove fakevar. I tried using var.ref but it dies on deserialization

Reply:
We can probably use `.ref()` instead of holding a FakeVar with name.